pydicom/filewriter.py

```diff
@@ -414,6 +414,17 @@ def write_data_element(fp, data_element, encoding=default_encoding):
     if (hasattr(data_element, "is_undefined_length")
             and data_element.is_undefined_length):
         is_undefined_length = True
+        # valid pixel data with undefined length shall contain encapsulated
+        # data, e.g. sequence items - raise ValueError otherwise (see #238)
+        if data_element.tag == 0x7fe00010:  # pixel data
```

Comment:
> I would probably renamed val to first_bytes

That would not be true - it is the whole value

>  It night be possible to cut onto 4 lines (e.g. after and and or statement.

I did that now - please check if that is ok for you.